crates/leon/benches/values.rs

```diff
@@ -0,0 +1,438 @@
+            black_box(
+                template
+                    .render(&vals(|_| Some(Cow::Borrowed("marcus"))))
+                    .unwrap(),
+            );
+        })
+    });
+    c.bench_function("one replace, hashmap", |b| {
+        b.iter(|| {
+            let template = Template::parse(black_box(TEMPLATE)).unwrap();
+            black_box(template.render(&hashmap).unwrap());
+        })
+    });
+    c.bench_function("one replace, slice", |b| {
+        b.iter(|| {
+            let template = Template::parse(black_box(TEMPLATE)).unwrap();
+            black_box(template.render(&slice).unwrap());
+        })
+    });
+}
+
+fn some_replaces(c: &mut Criterion) {
+    const TEMPLATE: &str = "hello {name}! i am {age} years old. my goal is to {goal}. i like: {flower}, {music}, {animal}, {color}, {food}. i'm drinking {drink}";
+    let mut hashmap = HashMap::new();
+    hashmap.insert("name", "marcus");
+    hashmap.insert("age", "42");
+    hashmap.insert("goal", "primary");
+    hashmap.insert("flower", "lotus");
+    hashmap.insert("music", "jazz");
+    hashmap.insert("animal", "cat");
+    hashmap.insert("color", "blue");
+    hashmap.insert("food", "pizza");
+    hashmap.insert("drink", "coffee");
+
+    let slice = &[
+        ("name", "marcus"),
+        ("age", "42"),
+        ("goal", "primary"),
+        ("flower", "lotus"),
+        ("music", "jazz"),
+        ("animal", "cat"),
+        ("color", "blue"),
+        ("food", "pizza"),
+        ("drink", "coffee"),
+    ];
+
+    c.bench_function("some replaces, fn", |b| {
+        b.iter(|| {
+            let template = Template::parse(black_box(TEMPLATE)).unwrap();
+            black_box(
+                template
+                    .render(&vals(|key| match key {
+                        "name" => Some(Cow::Borrowed("marcus")),
+                        "age" => Some(Cow::Borrowed("42")),
+                        "goal" => Some(Cow::Borrowed("primary")),
+                        "flower" => Some(Cow::Borrowed("lotus")),
+                        "music" => Some(Cow::Borrowed("jazz")),
+                        "animal" => Some(Cow::Borrowed("cat")),
+                        "color" => Some(Cow::Borrowed("blue")),
+                        "food" => Some(Cow::Borrowed("pizza")),
+                        "drink" => Some(Cow::Borrowed("coffee")),
+                        _ => None,
+                    }))
+                    .unwrap(),
+            );
+        })
+    });
+    c.bench_function("some replaces, hashmap", |b| {
+        b.iter(|| {
+            let template = Template::parse(black_box(TEMPLATE)).unwrap();
+            black_box(template.render(&hashmap).unwrap());
+        })
+    });
+    c.bench_function("some replaces, slice", |b| {
+        b.iter(|| {
+            let template = Template::parse(black_box(TEMPLATE)).unwrap();
+            black_box(template.render(&slice).unwrap());
+        })
+    });
+}
+
+fn many_replaces(c: &mut Criterion) {
```

Comment:
I wish we can have a function so that `some_replaces` and `many_replaces` can just call into with data and fn.